code/game/objects/items/devices/motion_detector.dm

```diff
@@ -70,26 +71,43 @@
 	blip_pool = list()
 	..()
 
+/obj/item/device/motiondetector/dropped(mob/user)
+	. = ..()
+	operator = null
+
+
 /obj/item/device/motiondetector/update_icon()
 	if(active)
 		icon_state = "detector_on_[detector_mode]"
 	else
 		icon_state = "detector_off"
 	return ..()
 
+/obj/item/device/motiondetector/proc/deactivate()
+	active = FALSE
+	operator = null
+	update_icon()
+	processing_objects.Remove(src)
+
 /obj/item/device/motiondetector/process()
 	if(!active)
-		update_icon()
-		processing_objects.Remove(src)
+		deactivate()
 		return
 
-	var/mob/living/carbon/human/human_user
-	if(ishuman(loc))
-		human_user = loc
-	else
-		active = FALSE
-		update_icon()
-		processing_objects.Remove(src)
+	if(!operator)
+		deactivate()
+		return
+
+	if(!loc == operator)
```

Comment:
```suggestion
	if(get_turf(src) != get_turf(operator))
```
Though all these checks could better go in a single `if` statement in between `||`